cvefeed/diff.go

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cvefeed
+
+import (
+	"encoding/json"
+	"math/bits"
+
+	"github.com/facebookincubator/nvdtools/cvefeed/nvd"
+	"github.com/facebookincubator/nvdtools/cvefeed/nvd/schema"
+)
+
+type bag map[string]interface{}
+
+// ChunkKind is the type of chunks produced by a diff.
+type ChunkKind string
+
+const (
+	// ChunkDescription indicates a difference in the description of a
+	// vulnerability.
+	ChunkDescription ChunkKind = "description"
+	// ChunkScore indicates a difference in the score of a vulnerability.
+	ChunkScore = "score"
+)
+
+type chunk uint32
+
+const (
+	chunkDescriptionShift = iota
+	chunkScoreShift
+	chunkMaxShift
+)
+
+const (
+	chunkDescription chunk = 1 << iota
+	chunkScore
+)
+
+var chunkKind = [chunkMaxShift]ChunkKind{
+	ChunkDescription,
+	ChunkScore,
+}
+
+func chunkShiftByName(kind ChunkKind) int {
+	for i, v := range chunkKind {
+		if v == kind {
+			return i
+		}
+	}
+	return chunkMaxShift
+}
+
+type diffEntry struct {
+	id   string
+	bits chunk
+}
+
+type diffFeed struct {
+	name string
+	dict Dictionary
+}
+
+type diffInput struct {
+	a, b diffFeed
+}
+
+type diff struct {
+	diffInput
+}
```

Comment:
I'm probably missing something out, but why is this a type?

Reply:
left over from some refactoring